src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,155 @@
+package duke;
+
+import javafx.application.Application;
+import javafx.geometry.Insets;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.layout.GridPane;
+import javafx.stage.Stage;
+
+/**
+ * Main class for running bot.
+ */
+public class Duke extends Application {
+    private static Label outputLabel;
+    private static final String FILE_PATH = (System.getProperty("user.dir").endsWith("text-ui-test")
+        ? System.getProperty("user.dir").substring(0, System.getProperty("user.dir").length() - 13)
+            + "/data/duke.txt"
+        : System.getProperty("user.dir") + "/data/duke.txt");
```

Comment:
Should this be abstracted to a separate method instead?

Reply:
Right, abstracting programming logic would make the code more clean.